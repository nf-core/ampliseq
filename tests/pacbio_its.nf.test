nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    profile "test_pacbio_its"

    test("-profile test_pacbio_its") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])

            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/software_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // Manually chosen files for content checks
                    path("$outputDir/overall_summary.tsv"),
                    path("$outputDir/barrnap/rrna.arc.gff"),
                    path("$outputDir/barrnap/rrna.bac.gff"),
                    path("$outputDir/barrnap/rrna.euk.gff"),
                    path("$outputDir/barrnap/rrna.mito.gff"),
                    path("$outputDir/cutadapt/cutadapt_summary.tsv"),
                    path("$outputDir/dada2/ASV_seqs.fasta"),
                    path("$outputDir/dada2/ASV_table.tsv"),
                    path("$outputDir/dada2/DADA2_stats.tsv"),
                    path("$outputDir/dada2/DADA2_table.rds"),
                    path("$outputDir/dada2/DADA2_table.tsv"),
                    path("$outputDir/input/Samplesheet_pacbio_ITS.tsv"),
                    path("$outputDir/input/Metadata_pacbio_ITS.tsv"),
                    path("$outputDir/itsx/ASV_ITS_seqs.full.fasta"),
                    path("$outputDir/itsx/ITSx.args.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt"),
                    path("$outputDir/SBDI/dna.tsv"),
                    path("$outputDir/SBDI/emof.tsv"),
                    path("$outputDir/SBDI/event.tsv")
                ).match() }
            )
        }
    }
}
