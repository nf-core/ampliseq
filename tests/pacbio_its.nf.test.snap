{
    "-profile test_pacbio_its": {
        "content": [
            34,
            {
                "BARRNAP": {
                    "barrnap": 0.9
                },
                "BARRNAPSUMMARY": {
                    "python": "Python 3.9.1"
                },
                "CUTADAPT_BASIC": {
                    "cutadapt": 4.6
                },
                "CUTADAPT_SUMMARY_STD": {
                    "python": "Python 3.8.3"
                },
                "DADA2_DENOISING": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_ERR": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_FILTNTRIM": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_MERGE": {
                    "R": "4.1.1",
                    "dada2": "1.22.0"
                },
                "DADA2_QUALITY1": {
                    "R": "4.3.2",
                    "dada2": "1.30.0",
                    "ShortRead": "1.60.0"
                },
                "DADA2_QUALITY2": {
                    "R": "4.3.2",
                    "dada2": "1.30.0",
                    "ShortRead": "1.60.0"
                },
                "DADA2_RMCHIMERA": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "DADA2_STATS": {
                    "R": "4.3.2",
                    "dada2": "1.30.0"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "FORMAT_TAXONOMY_SINTAX": {
                    "bash": "5.0.16"
                },
                "FORMAT_TAXRESULTS_SINTAX": {
                    "python": "3.9.1"
                },
                "ITSX_CUTASV": {
                    "ITSx": "1.1.3"
                },
                "MERGE_STATS_STD": {
                    "R": "4.3.2"
                },
                "RENAME_RAW_DATA_FILES": {
                    "sed": 4.7
                },
                "SBDIEXPORT": {
                    "R": "3.6.3"
                },
                "TREESUMMARIZEDEXPERIMENT": {
                    "R": "4.3.2",
                    "TreeSummarizedExperiment": "2.10.0"
                },
                "VSEARCH_SINTAX": {
                    "vsearch": "2.21.1"
                },
                "Workflow": {
                    "nf-core/ampliseq": "v2.15.0dev"
                }
            },
            [
                "SBDI",
                "SBDI/annotation.tsv",
                "SBDI/asv-table.tsv",
                "SBDI/dna.tsv",
                "SBDI/emof.tsv",
                "SBDI/event.tsv",
                "barrnap",
                "barrnap/rrna.arc.gff",
                "barrnap/rrna.bac.gff",
                "barrnap/rrna.euk.gff",
                "barrnap/rrna.mito.gff",
                "barrnap/summary.tsv",
                "cutadapt",
                "cutadapt/cutadapt_summary.tsv",
                "cutadapt/pb1.trimmed.cutadapt.log",
                "cutadapt/pb2.trimmed.cutadapt.log",
                "cutadapt/pb3.trimmed.cutadapt.log",
                "dada2",
                "dada2/ASV_seqs.fasta",
                "dada2/ASV_table.tsv",
                "dada2/DADA2_stats.tsv",
                "dada2/DADA2_table.rds",
                "dada2/DADA2_table.tsv",
                "dada2/QC",
                "dada2/QC/1.err.convergence.txt",
                "dada2/QC/1.err.pdf",
                "dada2/QC/3 files were used for single_end plotQualityProfile.txt",
                "dada2/QC/single_end_preprocessed_qual_stats.pdf",
                "dada2/QC/single_end_qual_stats.pdf",
                "dada2/QC/svg",
                "dada2/QC/svg/1.err.svg",
                "dada2/QC/svg/single_end_preprocessed_qual_stats.svg",
                "dada2/QC/svg/single_end_qual_stats.svg",
                "dada2/args",
                "dada2/args/dada.args.txt",
                "dada2/args/filterAndTrim.args.txt",
                "dada2/args/learnErrors.args.txt",
                "dada2/args/removeBimeraDenovo.args.txt",
                "dada2/args/single_end_plotQualityProfile.args.txt",
                "dada2/args/single_end_preprocessed_plotQualityProfile.args.txt",
                "dada2/log",
                "dada2/log/1.dada.log",
                "dada2/log/1.err.log",
                "fastqc",
                "fastqc/pb1_fastqc.html",
                "fastqc/pb2_fastqc.html",
                "fastqc/pb3_fastqc.html",
                "input",
                "input/Metadata_pacbio_ITS.tsv",
                "input/Samplesheet_pacbio_ITS.tsv",
                "itsx",
                "itsx/ASV_ITS_seqs.full.fasta",
                "itsx/ASV_ITS_seqs.summary.txt",
                "itsx/ASV_len_filt.tsv",
                "itsx/ASV_len_orig.tsv",
                "itsx/ASV_seqs.len.fasta",
                "itsx/ASV_table.len.tsv",
                "itsx/ITSx.args.txt",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/cutadapt_filtered_reads_plot.txt",
                "multiqc/multiqc_data/cutadapt_trimmed_sequences_plot_3_Counts.txt",
                "multiqc/multiqc_data/cutadapt_trimmed_sequences_plot_3_Obs_Exp.txt",
                "multiqc/multiqc_data/cutadapt_trimmed_sequences_plot_5_Counts.txt",
                "multiqc/multiqc_data/cutadapt_trimmed_sequences_plot_5_Obs_Exp.txt",
                "multiqc/multiqc_data/fastqc-status-check-heatmap.txt",
                "multiqc/multiqc_data/fastqc_adapter_content_plot.txt",
                "multiqc/multiqc_data/fastqc_overrepresented_sequences_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_counts_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt",
                "multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_cutadapt.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_fastqc.txt",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_plots/pdf",
                "multiqc/multiqc_plots/pdf/cutadapt_filtered_reads_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/cutadapt_filtered_reads_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/cutadapt_trimmed_sequences_plot_3_Counts.pdf",
                "multiqc/multiqc_plots/pdf/cutadapt_trimmed_sequences_plot_3_Obs_Exp.pdf",
                "multiqc/multiqc_plots/pdf/cutadapt_trimmed_sequences_plot_5_Counts.pdf",
                "multiqc/multiqc_plots/pdf/cutadapt_trimmed_sequences_plot_5_Obs_Exp.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-status-check-heatmap.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_adapter_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_overrepresented_sequences_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_n_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_sequence_quality_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_quality_scores_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_duplication_levels_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_length_distribution_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_top_overrepresented_sequences_table.pdf",
                "multiqc/multiqc_plots/pdf/general_stats_table.pdf",
                "multiqc/multiqc_plots/png",
                "multiqc/multiqc_plots/png/cutadapt_filtered_reads_plot-cnt.png",
                "multiqc/multiqc_plots/png/cutadapt_filtered_reads_plot-pct.png",
                "multiqc/multiqc_plots/png/cutadapt_trimmed_sequences_plot_3_Counts.png",
                "multiqc/multiqc_plots/png/cutadapt_trimmed_sequences_plot_3_Obs_Exp.png",
                "multiqc/multiqc_plots/png/cutadapt_trimmed_sequences_plot_5_Counts.png",
                "multiqc/multiqc_plots/png/cutadapt_trimmed_sequences_plot_5_Obs_Exp.png",
                "multiqc/multiqc_plots/png/fastqc-status-check-heatmap.png",
                "multiqc/multiqc_plots/png/fastqc_adapter_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_overrepresented_sequences_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_n_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_sequence_quality_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Counts.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_quality_scores_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_duplication_levels_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_length_distribution_plot.png",
                "multiqc/multiqc_plots/png/fastqc_top_overrepresented_sequences_table.png",
                "multiqc/multiqc_plots/png/general_stats_table.png",
                "multiqc/multiqc_plots/svg",
                "multiqc/multiqc_plots/svg/cutadapt_filtered_reads_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/cutadapt_filtered_reads_plot-pct.svg",
                "multiqc/multiqc_plots/svg/cutadapt_trimmed_sequences_plot_3_Counts.svg",
                "multiqc/multiqc_plots/svg/cutadapt_trimmed_sequences_plot_3_Obs_Exp.svg",
                "multiqc/multiqc_plots/svg/cutadapt_trimmed_sequences_plot_5_Counts.svg",
                "multiqc/multiqc_plots/svg/cutadapt_trimmed_sequences_plot_5_Obs_Exp.svg",
                "multiqc/multiqc_plots/svg/fastqc-status-check-heatmap.svg",
                "multiqc/multiqc_plots/svg/fastqc_adapter_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_overrepresented_sequences_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_n_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_sequence_quality_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_duplication_levels_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_length_distribution_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_top_overrepresented_sequences_table.svg",
                "multiqc/multiqc_plots/svg/general_stats_table.svg",
                "multiqc/multiqc_report.html",
                "overall_summary.tsv",
                "pipeline_info",
                "pipeline_info/software_versions.yml",
                "sintax",
                "sintax/ASV_ITS_tax_sintax.unite-fungi_8_2.raw.tsv",
                "sintax/ASV_tax_sintax.unite-fungi_8_2.tsv",
                "sintax/ref_taxonomy_sintax.txt",
                "summary_report",
                "summary_report/itsx_preliminary_origin.svg",
                "summary_report/rrna_detection_with_barrnap.svg",
                "summary_report/sintax_taxonomic_classification_per_taxonomy_level.svg",
                "summary_report/stacked_barchart_of_reads.svg",
                "summary_report/summary_report.html",
                "summary_report/versions.yml",
                "treesummarizedexperiment",
                "treesummarizedexperiment/sintax_TreeSummarizedExperiment.rds"
            ],
            "overall_summary.tsv:md5,97a8ee12c40a0bc582ba1cefda117f95",
            "rrna.arc.gff:md5,6bb50a4c244d80d56e759ecfbb045381",
            "rrna.bac.gff:md5,7c38dd6692b792ace17142158afeb33e",
            "rrna.euk.gff:md5,8ae155cbe43335084a0c9a2ccd655cb3",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf",
            "cutadapt_summary.tsv:md5,d185ccd02c22b859fcaf2e5158ce70c6",
            "ASV_seqs.fasta:md5,a2632b1cc533d290db19ceedaf350dcd",
            "ASV_table.tsv:md5,6de5507a0fcb0c9b03f8fec0270440be",
            "DADA2_stats.tsv:md5,4bbde0bbf579f60e692bd32d14034f0d",
            "DADA2_table.rds:md5,a36558c04a7b21050bb6e479b09bd483",
            "DADA2_table.tsv:md5,7e7aab7ae9a3b20fb08f3fe7f8b0430c",
            "Samplesheet_pacbio_ITS.tsv:md5,905b0fc4850149d6457c9da7076c2c2d",
            "Metadata_pacbio_ITS.tsv:md5,51d81b0fe29f6377fa21ce8399fc5ae0",
            "ASV_ITS_seqs.full.fasta:md5,dab051f759c877a79adf47f5729f5665",
            "ITSx.args.txt:md5,32430f2c36d83c08273352bad45857e8",
            "multiqc_fastqc.txt:md5,f11936dbbf1718301ceba86050e24799",
            "multiqc_general_stats.txt:md5,1fef8358fe3cc974bf05e2f27fc7e250",
            "multiqc_cutadapt.txt:md5,ae5e5bf06cdbfa4d8305bbcb61c3ba04",
            "dna.tsv:md5,061ca46725294da75acd8237390f4291",
            "emof.tsv:md5,c6599c76a70fe5458db56ff8c2e89a37",
            "event.tsv:md5,bfcf7d8ac2a3c6b48d075d6dc235f0e4"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-12T16:37:37.033224088"
    }
}